### PR TITLE
Bring Data in SCLF into sync with the version in the overlay

### PR DESCRIPTION
**Pull request by saiHemak**

### CODE REVIEW

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
CC: @phausler

**Comment by alblue** [comment]
Typo in commit message (alining -> aligning) if you can amend the commit message.

File: Foundation/Data.swift [hunk 1]
```diff
@@ -1553,7 +1558,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     /// - parameter buffer: The replacement bytes.
     @inline(__always)
     public mutating func replaceSubrange<SourceType>(_ subrange: Range<Index>, with buffer: UnsafeBufferPointer<SourceType>) {
-        guard !buffer.isEmpty else { return }
```
Comment (ikesyo) [inline Foundation/Data.swift]: Please don’t revert the changes in https://github.com/apple/swift-corelibs-foundation/pull/1485.

Comment (ianpartridge) [reply Foundation/Data.swift]: @saiHemak what happened here? I thought you were taking the latest version of Data from the overlay?

**Comment by saiHemak** [comment]
@ianpartridge I corrected it .. 
@parkera  Please review

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
Can we get a review from someone appropriate - @phausler @parkera ?

**Comment by parkera** [comment]
@saiHemak this is just copied from the overlay, no additional changes?

File: Foundation/Data.swift [hunk 2]
```diff
@@ -1474,7 +1478,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     /// - parameter buffer: The buffer of bytes to append. The size is calculated from `SourceType` and `buffer.count`.
     @inline(__always)
     public mutating func append<SourceType>(_ buffer : UnsafeBufferPointer<SourceType>) {
-        if buffer.isEmpty { return }
+        if buffer.count == 0 { return }
```
Comment (ianpartridge) [inline Foundation/Data.swift]: This is wrong: https://github.com/apple/swift/blob/master/stdlib/public/SDK/Foundation/Data.swift#L1467

Comment (saiHemak) [reply Foundation/Data.swift]: Please refer to @ikesyo 's  comment related to PR https://github.com/apple/swift-corelibs-foundation/pull/1485/files.This change has been implemented as part of gardening work done by @ikesyo   and I have picked those changes from latest master of SCLF.

**Comment by saiHemak** [comment]
@ianpartridge  ikesyo's changes were brought in to Swift Overlay recently. Initially the overlay had count changes .Hence ,initially when the PR was raised the latest master of Overlay had Count changes which caused the confusion.

**Comment by saiHemak** [comment]
@ianpartridge a day ago some more changes seems to be added to Data.swift am picking up those too ..

File: Foundation/Data.swift (hunk 2 above)
Comment (saiHemak) [reply Foundation/Data.swift]: :) ..Yeah anyway rebasing based on latest master

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
@parkera looks good to me, your or @phausler 's OK would be welcome.
